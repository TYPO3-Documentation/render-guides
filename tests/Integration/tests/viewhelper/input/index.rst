=================
Fluid ViewHelpers
=================

See :typo3:viewhelper:`typo3fluid-fluid-viewhelpers-splitviewhelper` or
:typo3:viewhelper-argument:`typo3-cms-fluid-viewhelpers-link-externalviewhelper-uri`.

f:split
=======

..  typo3:viewhelper:: split
    :source: resources/global_viewhelpers_demo.json

..  typo3:viewhelper:: split
    :source: resources/global_viewhelpers_demo.json
    :sortBy: json
    :noindex:

f:link.external
===============

..  typo3:viewhelper:: link.external
    :source: resources/global_viewhelpers_demo.json

f:deprecated
============

..  typo3:viewhelper:: deprecated
    :source: resources/global_viewhelpers_demo.json


formvh:be.maximumFileSize
=========================

..  typo3:viewhelper:: be.maximumFileSize
    :source: resources/Form.json
