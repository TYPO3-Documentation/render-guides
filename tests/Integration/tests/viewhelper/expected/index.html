<!-- content start -->
                <section class="section" id="fluid-viewhelpers">
            <h1>Fluid ViewHelpers<a class="headerlink" href="#fluid-viewhelpers" data-bs-toggle="modal" data-bs-target="#linkReferenceModal"  title="Reference this headline">¶</a></h1>
            <section class="section" id="f-split">
            <h2>f:split<a class="headerlink" href="#f-split" data-bs-toggle="modal" data-bs-target="#linkReferenceModal"  title="Reference this headline">¶</a></h2>


    <p>The SplitViewHelper splits a string by the specified separator, which
results in an array. The number of values in the resulting array can
be limited with the limit parameter, which results in an array where
the last item contains the remaining unsplit string.
This ViewHelper mimicks PHP&#039;s <code class="code-inline" translate="no" aria-description="PHP" aria-details="Dynamic server-side scripting language.">explode()</code> function.</p>
<section class="section" id="examples">
            <h2>Examples<a class="headerlink" href="#examples" data-bs-toggle="modal" data-bs-target="#linkReferenceModal"  title="Reference this headline">¶</a></h2>
            <section class="section" id="split-with-a-separator">
            <h3>Split with a separator<a class="headerlink" href="#split-with-a-separator" data-bs-toggle="modal" data-bs-target="#linkReferenceModal"  title="Reference this headline">¶</a></h3>
            <div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code>&lt;f:split value=&quot;1,5,8&quot; separator=&quot;,&quot; /&gt;</code></pre>
    <div>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>
            <div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code class="hljs plaintext">{0: '1', 1: '5', 2: '8'}</code></pre>
    <div>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>
    </section>
            <section class="section" id="split-using-tag-content-as-value">
            <h3>Split using tag content as value<a class="headerlink" href="#split-using-tag-content-as-value" data-bs-toggle="modal" data-bs-target="#linkReferenceModal"  title="Reference this headline">¶</a></h3>
            <div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code>&lt;f:split separator=&quot;-&quot;&gt;1-5-8&lt;/f:split&gt;</code></pre>
    <div>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>
            <div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code class="hljs plaintext">{0: '1', 1: '5', 2: '8'}</code></pre>
    <div>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>
    </section>
            <section class="section" id="split-with-a-limit">
            <h3>Split with a limit<a class="headerlink" href="#split-with-a-limit" data-bs-toggle="modal" data-bs-target="#linkReferenceModal"  title="Reference this headline">¶</a></h3>
            <div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code>&lt;f:split value=&quot;1,5,8&quot; separator=&quot;,&quot; limit=&quot;2&quot; /&gt;</code></pre>
    <div>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>
            <div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code class="hljs plaintext">{0: '1', 1: '5,8'}</code></pre>
    <div>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>
    </section>
    </section>

    <h2>Source code</h2>

<p>
    Go to the source code of this ViewHelper:
    <a href="https://github.com/TYPO3/Fluid/blob/main/src/ViewHelpers/SplitViewHelper.php">SplitViewHelper.php (GitHub)</a>.
</p>

    <h2>Arguments</h2>
    <p>The following arguments are available for the split ViewHelper: </p>
            <dl class="confval">
    <dt id="viewhelper-argument-typo3fluid-fluid-viewhelpers-splitviewhelper-value" class="d-flex justify-content-between">
        <div class="confval-header">
            <code class="sig-name descname"><span class="pre">value</span></code>
                            <a class="headerlink" href="#viewhelper-argument-typo3fluid-fluid-viewhelpers-splitviewhelper-value" data-bs-toggle="modal" data-bs-target="#linkReferenceModal" data-id="viewhelper-argument-typo3fluid-fluid-viewhelpers-splitviewhelper-value" data-rstCode=""  title="Reference this configuration value">¶</a>
            </div>
        <div class="confval-back-to-top">                <a href="#" class="backToTop" title="Back to top"><i class="fa-solid fa-arrow-up fa-xs"></i></a>        </div>
    </dt>
    <dd>
        <dl class="field-list simple">
            <dt class="field">Type</dt>
            <dd class="field">string
            </dd>
                            <dt class="field">Default</dt>
                <dd class="field">&#039;&#039;
                </dd>
                                </dl>
        <div class="confval-description">
            The string to explode
        </div>
    </dd>
</dl>            <dl class="confval">
    <dt id="viewhelper-argument-typo3fluid-fluid-viewhelpers-splitviewhelper-separator" class="d-flex justify-content-between">
        <div class="confval-header">
            <code class="sig-name descname"><span class="pre">separator</span></code>
                            <a class="headerlink" href="#viewhelper-argument-typo3fluid-fluid-viewhelpers-splitviewhelper-separator" data-bs-toggle="modal" data-bs-target="#linkReferenceModal" data-id="viewhelper-argument-typo3fluid-fluid-viewhelpers-splitviewhelper-separator" data-rstCode=""  title="Reference this configuration value">¶</a>
            </div>
        <div class="confval-back-to-top">                <a href="#" class="backToTop" title="Back to top"><i class="fa-solid fa-arrow-up fa-xs"></i></a>        </div>
    </dt>
    <dd>
        <dl class="field-list simple">
            <dt class="field">Type</dt>
            <dd class="field">string
            </dd>
                                        <dt class="field">Required</dt>
                <dd class="field">1
                </dd>
                    </dl>
        <div class="confval-description">
            Separator string to explode with
        </div>
    </dd>
</dl>            <dl class="confval">
    <dt id="viewhelper-argument-typo3fluid-fluid-viewhelpers-splitviewhelper-limit" class="d-flex justify-content-between">
        <div class="confval-header">
            <code class="sig-name descname"><span class="pre">limit</span></code>
                            <a class="headerlink" href="#viewhelper-argument-typo3fluid-fluid-viewhelpers-splitviewhelper-limit" data-bs-toggle="modal" data-bs-target="#linkReferenceModal" data-id="viewhelper-argument-typo3fluid-fluid-viewhelpers-splitviewhelper-limit" data-rstCode=""  title="Reference this configuration value">¶</a>
            </div>
        <div class="confval-back-to-top">                <a href="#" class="backToTop" title="Back to top"><i class="fa-solid fa-arrow-up fa-xs"></i></a>        </div>
    </dt>
    <dd>
        <dl class="field-list simple">
            <dt class="field">Type</dt>
            <dd class="field">int
            </dd>
                            <dt class="field">Default</dt>
                <dd class="field">0
                </dd>
                                </dl>
        <div class="confval-description">
            If limit is positive, a maximum of $limit items will be returned. If limit is negative, all items except for the last $limit items will be returned. 0 will be treated as 1.
        </div>
    </dd>
</dl>
    </section>
            <section class="section" id="f-link-external">
            <h2>f:link.external<a class="headerlink" href="#f-link-external" data-bs-toggle="modal" data-bs-target="#linkReferenceModal"  title="Reference this headline">¶</a></h2>


    <p>A ViewHelper for creating links to external targets.</p>
<section class="section" id="examples">
            <h2>Examples<a class="headerlink" href="#examples" data-bs-toggle="modal" data-bs-target="#linkReferenceModal"  title="Reference this headline">¶</a></h2>
            <section class="section" id="default">
            <h3>Default<a class="headerlink" href="#default" data-bs-toggle="modal" data-bs-target="#linkReferenceModal"  title="Reference this headline">¶</a></h3>
            <div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code>&lt;f:link.external uri=&quot;https://www.typo3.org&quot; target=&quot;_blank&quot;&gt;external link&lt;/f:link.external&gt;
</code></pre>
    <div>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>

    <p>Output:</p>

            <div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code>&lt;a href=&quot;https://www.typo3.org&quot; target=&quot;_blank&quot;&gt;external link&lt;/a&gt;
</code></pre>
    <div>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>
    </section>
            <section class="section" id="custom-default-scheme">
            <h3>Custom default scheme<a class="headerlink" href="#custom-default-scheme" data-bs-toggle="modal" data-bs-target="#linkReferenceModal"  title="Reference this headline">¶</a></h3>
            <div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code>&lt;f:link.external uri=&quot;typo3.org&quot; defaultScheme=&quot;ftp&quot;&gt;external ftp link&lt;/f:link.external&gt;
</code></pre>
    <div>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>

    <p>Output:</p>

            <div class="code-block-wrapper" translate="no">

    <pre class="code-block"><code>&lt;a href=&quot;ftp://typo3.org&quot;&gt;external ftp link&lt;/a&gt;</code></pre>
    <div>        <button class="code-block-copy" title="Copy to clipboard">
            <i class="icon fa-regular fa-copy code-block-copy-icon"></i>
            <i class="fa-solid fa-check code-block-check-icon code-block-hide"></i>
        </button>
        <span class="code-block-check-tooltip code-block-hide" aria-live="polite">Copied!</span>
    </div>
</div>
    </section>
    </section>

    <h2>Source code</h2>

<p>
    Go to the source code of this ViewHelper:
    <a href="https://github.com/TYPO3/typo3/blob/main/typo3/sysext/fluid/Classes/ViewHelpers/Link/ExternalViewHelper.php">Link\ExternalViewHelper.php (GitHub)</a>.
</p>

    <h2>Arguments</h2>
    <p>The following arguments are available for the link.external ViewHelper: </p>
            <dl class="confval">
    <dt id="viewhelper-argument-typo3-cms-fluid-viewhelpers-link-externalviewhelper-uri" class="d-flex justify-content-between">
        <div class="confval-header">
            <code class="sig-name descname"><span class="pre">uri</span></code>
                            <a class="headerlink" href="#viewhelper-argument-typo3-cms-fluid-viewhelpers-link-externalviewhelper-uri" data-bs-toggle="modal" data-bs-target="#linkReferenceModal" data-id="viewhelper-argument-typo3-cms-fluid-viewhelpers-link-externalviewhelper-uri" data-rstCode=""  title="Reference this configuration value">¶</a>
            </div>
        <div class="confval-back-to-top">                <a href="#" class="backToTop" title="Back to top"><i class="fa-solid fa-arrow-up fa-xs"></i></a>        </div>
    </dt>
    <dd>
        <dl class="field-list simple">
            <dt class="field">Type</dt>
            <dd class="field">string
            </dd>
                                        <dt class="field">Required</dt>
                <dd class="field">1
                </dd>
                    </dl>
        <div class="confval-description">
            The URI that will be put in the href attribute of the rendered link tag
        </div>
    </dd>
</dl>
    </section>
            <section class="section" id="f-deprecated">
            <h2>f:deprecated<a class="headerlink" href="#f-deprecated" data-bs-toggle="modal" data-bs-target="#linkReferenceModal"  title="Reference this headline">¶</a></h2>
                                    <div class="versionchange deprecated">
                <p class="versionmodified">
                    <span class="versionicon"><i class="fa-solid fa-ban"></i></span> Deprecated </p>
                <article>
                    since v11, will be removed in v12
                </article>
            </div>


    <p>Example for deprecated ViewHelper and complex types</p>


    <h2>Source code</h2>

<p>
    Go to the source code of this ViewHelper:
    <a href="https://github.com/TYPO3/typo3/blob/main/typo3/sysext/fluid/Classes/ViewHelpers/DeprecatedViewHelper.php">DeprecatedViewHelper.php (GitHub)</a>.
</p>


    </section>
    </section>
        <!-- content end -->
